### PR TITLE
Add documentation for the Data Model

### PR DESCRIPTION
**Pull request by jguertin**
While I was in there I created a missing relationship for registration payments

### CODE REVIEW

File: ppr-api/docs/data_model/README.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+Database setup is meant to be handled by Alembic scripts which are executed when a PPR API pod starts up. This ensures
+that the database state will always be consistent with the API implementation.
+
+There are a few core concepts in the data model, described in the next sections. 
+
+### Financing Statements
+
+Financing Statements have a hierarchical structure rooted in the `financing_statement` table. There are 2 ways, access
+a financing statement: Current state quick lookup and historical event based rebuild.
```
Comment (WalterMoar) [inline ppr-api/docs/data_model/README.md]: Is there a word missing here? "There are 2 ways *to* access..."?

File: ppr-api/docs/data_model/README.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+At the core, updates to financing statements need to be able to reflect the state at a particular time, so it is
+important that any changes be non-destructive.  This is why some data may appear to be duplicated, so it is correct in
+both contexts.
+
+#### Event Based Data Lookup
+
+Data that can be change at the root of a financing statement is stored in the `registration` table. Withing the API and
```
Comment (WalterMoar) [inline ppr-api/docs/data_model/README.md]: - "Data that can be change*d*..."?
- "With*in* the API..."

File: ppr-api/docs/data_model/README.md [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+project vocabulary, a `registration` record is interchangeable with the term "**Event**". In the case of `life`, if the
+event is for a renewal, then the length of the renewal (or `-1` if infinite) is stored. Events that don't affect life
+would have a `null` value in `life`.
+
+For entities that are related to financing statements (Parties and Collateral), each table will have 2 foreign keys to
+`registration`: a starting event reference and an ending event reference. Changes to these tables would be handled by
+"start/end pairs" rather than editing the data.
+
+For example, if and end user corrected a typo in an address for a financing statement, rather than modify the address a
```
Comment (WalterMoar) [inline ppr-api/docs/data_model/README.md]: "...if _an_ end user..."

File: ppr-api/docs/data_model/README.md [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+new `registration` event should be created.  This would do the following:
+- Insert a new `registration` associated to the `financing_statement`
+- Update the `fs_party.reg_number_end` of the row that belongs to the financing statement and owns the affected address.
+- Insert a new `fs_party` record as well as an associated `address`, setting `fs_party.reg_number_start` to the value of
+  `registration.reg_number`
+
+#### Current State
+
+For a quick lookup, you should be able to get the current state of a financing statement with processing each individual
+event. Core financing statement values should be stored on the `financing_statement` table. In the case of `life`, the
```
Comment (WalterMoar) [inline ppr-api/docs/data_model/README.md]: Should this say "..._without_ processing each individual event."?

File: ppr-api/docs/data_model/README.md (hunk 2 above)
Comment (jguertin) [reply ppr-api/docs/data_model/README.md]: Yeah, it's a bit of a weird one.  I'm hoping the example explains it a bit better.  Changed it to "Modifiable data that exists at the root of a financing statement ..."

Still a bit off, but maybe a bit better.

File: ppr-api/docs/data_model/README.md (hunk 1 above)
Comment (jguertin) [reply ppr-api/docs/data_model/README.md]: yes

File: ppr-api/docs/data_model/README.md (hunk 4 above)
Comment (jguertin) [reply ppr-api/docs/data_model/README.md]: Yes it should (facepalm)

**Review by WalterMoar — COMMENTED** [review]
Some minor typos, although a possible with/without switch could be misleading.

**Review by WalterMoar — APPROVED** [review]

**Pull request merged**